### ISSUE TITLE
TypeError: OnnxStableDiffusionPipeline.__init__() missing 1 required positional argument: 'vae_encoder'

### ISSUE THREAD

**Issue by kamalasubha** [huggingface/diffusers#1335]
### Describe the bug

Hi,
I tried ONNX Runtime for inference.  The code is,
`from diffusers import StableDiffusionOnnxPipeline
pipe = StableDiffusionOnnxPipeline.from_pretrained(
    "CompVis/stable-diffusion-v1-4",
    revision="onnx",
    provider="CPUExecutionProvider",
    use_auth_token=True,
) 
prompt = "a photo of an astronaut riding a horse on mars"
image = pipe(prompt).images[0]`

**Error is,
Traceback (most recent call last):
  File "Python-work\stable_diffuser_onnx_compvix.py", line 3, in <module>
    pipe = StableDiffusionOnnxPipeline.from_pretrained(
  File "onnx-virtual\lib\site-packages\diffusers\pipeline_utils.py", line 647, in from_pretrained
    model = pipeline_class(**init_kwargs)
  File "onnx-virtual\lib\site-packages\diffusers\pipelines\stable_diffusion\pipeline_onnx_stable_diffusion.py", line 272, in __init__
    super().__init__(
TypeError: OnnxStableDiffusionPipeline.__init__() missing 1 required positional argument: 'vae_encoder'**

Kindly help me to fix this error

### Reproduction

I tried this with virtual environment and python 3.10

### System Info

Environment is windows, python 3.10

**Comment by averad** [comment]
I use the Onnx pipeline with an AMD Card following the steps [here](https://gist.github.com/averad/256c507baa3dcc9464203dc14610d674)

I replaced `provider="DmlExecutionProvider"` with `provider="CPUExecutionProvider"` and was able to generate an image using my CPU.

Note: You will need to go to https://huggingface.co/CompVis/stable-diffusion-v1-4 and accept the model use terms before attempting to use the model or download it.

**Comment by MCRusher** [comment]
Yeah I had to give it explicitly with

`encoder = OnnxRuntimeModel.from_pretrained(model / "vae_encoder", provider=provider, sess_options=so)`

Not sure why it doesn't automatically pull it like it does the other ones.

**Comment by averad** [comment]
If you have time can you run `diffusers-cli env` and post the output

**Comment by kamalasubha** [comment]
Hi,
Please find the output
- `diffusers` version: 0.7.2
- Platform: Windows-10-10.0.19044-SP0
- Python version: 3.10.2
- PyTorch version (GPU?): 1.13.0+cpu (False)
- Huggingface_hub version: 0.11.0
- Transformers version: 4.24.0
- Using GPU in script?: <fill in>
- Using distributed or parallel set-up in script?: <fill in>

**Comment by averad** [comment]
@kamalasubha I was unable to duplicate the reported issue. Here are the steps I completed to attempt to replicate the issue:

1. pip install virtualenv
2. python -m venv sd_env
3. sd_env\scripts\activate
4. python -m pip install --upgrade pip
5. pip install git+https://github.com/huggingface/diffusers.git
6. pip install transformers onnxruntime onnx torch ftfy spacy
7. pip install onnxruntime-directml --force-reinstall
8. pip install protobuf==3.20.1
9. Accepted the Model License Agreement
10. Ran the following code to generate and image

```
from diffusers import OnnxStableDiffusionPipeline
height=512
width=512
num_inference_steps=50
guidance_scale=7.5
prompt = "a photo of an astronaut riding a horse on mars"
negative_prompt="bad hands, blurry"
pipe = OnnxStableDiffusionPipeline.from_pretrained("CompVis/stable-diffusion-v1-4", revision="onnx", provider="DmlExecutionProvider")
image = pipe(prompt, height, width, num_inference_steps, guidance_scale, negative_prompt).images[0] 
image.save("astronaut_rides_horse.png")
```

**Comment by patrickvonplaten** [comment]
cc @anton-l

**Comment by Justanetizen** [comment]
I have the same problem if I try to use [wd-v1-3-full.ckpt](https://huggingface.co/hakurei/waifu-diffusion-v1-3/blob/main/wd-v1-3-full.ckpt) converted to onnx with this script [Convert Original Stable Diffusion to Diffusers script](https://github.com/huggingface/diffusers/blob/main/scripts/convert_original_stable_diffusion_to_diffusers.py)

If I use https://huggingface.co/runwayml/stable-diffusion-v1-5 with the onnx branch then it works as expected.

**Comment by GreenLandisaLie** [comment]
@Justanetizen
The first script simply converts an original SD checkpoint to be compatible with diffusers. Its still a pytorch model hence the need to use the onnx converter script as you mentioned.

I believe 'revision="onnx"' will only work on models with a onnx branch and many do not have it - last time I checked - so the recommended approach for onnx should be to use the convert script. 
One thing to note is that initially the convert script did not create an onnx vae_encoder (or was it decoder?) and so anyone using an outdated version of the script will get the missing positional argument error - my guess is the script and onnx pipelines were updated for this but whatever code runs when using 'revision' - was not.

**Comment by averad** [comment]
@Justanetizen [Convert Original Stable Diffusion to Diffusers script](https://github.com/huggingface/diffusers/blob/main/scripts/convert_original_stable_diffusion_to_diffusers.py) doesn't convert ckpt files to Onnx.

To covert a ckpt file to onnx you have to run [Convert Original Stable Diffusion to Diffusers script](https://github.com/huggingface/diffusers/blob/main/scripts/convert_original_stable_diffusion_to_diffusers.py) and then [Convert Stable Diffusion Checkpoint to Onnx script](https://github.com/huggingface/diffusers/blob/main/scripts/convert_stable_diffusion_checkpoint_to_onnx.py) on the resulting diffusers model folder.

**Comment by GreenLandisaLie** [comment]
StableDiffusionOnnxPipeline is deprecated, you should use OnnxStableDiffusionPipeline instead. 
Also I'm unable to replicate this issue using an updated version of diffusers. Maybe this has already been solved? 
It might also be worth trying to delete the model cache stored in ...\{USER}\\.cache\huggingface\diffusers\models--CompVis--stable-diffusion-v1-4 and re run the script to grab the whole thing again - vae_encoder was not an argument in the earlier versions of the onnx pipeline (I think) so it might be possible that the pipe doesn't try to download a missing vae_encoder - which would indeed be a compatibility issue bug.

**Comment by kamalasubha** [comment]
@averad I am able to replicate the same steps and able to generate the image. Thanks
But, every time the model is getting downloaded.
In order to avoid it, I downloaded the model and convert to onnx type as per the comments given by you.
```
python convert_original_stable_diffusion_to_diffusers.py --checkpoint_path="./eldenring-v2-pruned.ckpt" --dump_path="./eldenring_v2_pruned_diffusers"
python convert_stable_diffusion_checkpoint_to_onnx.py --model_path="./eldenring_v2_pruned_diffusers" --output_path="./eldenring_v2_pruned_onnx"

```
When I am running with the above code,
`pipe = OnnxStableDiffusionPipeline.from_pretrained("./model_onnx", revision="onnx", provider="CPUExecutionProvider")
`
It throws me the error like,
```
return self._sess.run(output_names, input_feed, run_options)
onnxruntime.capi.onnxruntime_pybind11_state.InvalidArgument: [ONNXRuntimeError] : 2 : INVALID_ARGUMENT : Unexpected input data type. Actual: (tensor(int64)) , expected: (tensor(float))
```
@averad @GreenLandisaLie @Justanetizen 
Can you please help on the same?

**Comment by GreenLandisaLie** [comment]
@kamalasubha 
Don't use revision="onnx" - you are already calling from an onnx pipeline! And also you might want to install ort nightly directml and use DmlExecutionProvider instead of CPUExecutionProvider - it works on my RX 560 4G although it uses RAM as shared memory (which slows things quite a bit) to compensate the lack of VRAM - still 5 times faster than my CPU. If your GPU has 4G or more then you def should use DmlExecutionProvider.
Take a look at this [issue](https://github.com/huggingface/diffusers/issues/1316) I opened as it contains the link for it as well as a fix to a possible problem you might encounter (so far I seem to be the only one though) - don't forget you must grab the version specific for your python version.

**Comment by anton-l** [comment]
@kamalasubha this issue should be fixed in `diffusers>=0.8`. Try installing the latest version, then this will work:

```python
from diffusers import StableDiffusionOnnxPipeline 

pipe = StableDiffusionOnnxPipeline.from_pretrained("CompVis/stable-diffusion-v1-4", revision="onnx", provider="CPUExecutionProvider")
```
